frontend/src/metabase/visualizations/lib/settings/column.js

```diff
@@ -25,11 +25,8 @@ import { currency } from "cljs/metabase.shared.util.currency";
 const DEFAULT_GET_COLUMNS = (series, vizSettings) =>
   [].concat(...series.map(s => (s.data && s.data.cols) || []));
 
-export function columnSettings({
-  getColumns = DEFAULT_GET_COLUMNS,
-  hidden,
-  ...def
-} = {}) {
+export function columnSettings(args = {}) {
+  const { getColumns = DEFAULT_GET_COLUMNS, hidden, ...def } = args;
```

Comment:
TS somehow infers `getColumns: never[]` unless I destructure the args outside the function signature.  I’m not sure why.

Reply:
It looks like TS had trouble parsing the type of `[].concat(...series.map(...))` but it worked when I used `series.flatMap`.  I’ll make this change

Reply:
Do you still plan on changing this?